### ISSUE TITLE
Is support of LTS Symfony version needed?

### ISSUE THREAD

**Issue by FluffyDiscord** [FluffyDiscord/roadrunner-symfony-bundle#9]
If anyone needs this package to support LTS versions of Symfony (6.4) explicitly, let me know.

I am currently aiming to support and test only the latest stable.

**Comment by moshkov-konstantin** [comment]
Only 7.2, only hardcore =)

**Comment by FluffyDiscord** [comment]
> Only 7.2, only hardcore =)

7.2 should be working fine (not aware of any breaking changes), but I can run tests against it later today and release "official" version

**Comment by Zylius** [comment]
Hello could you please add support for 5.4 and 6.4? older bigger legacy projects sure needs this. I know 5.4 isnt in LTS anymore, but it still is used a lot, IDK why would you want to limit your user base because of a version which changes nothing of significance AFAIK.

**Comment by FluffyDiscord** [comment]
> Hello could you please add support for 5.4 and 6.4? older bigger legacy projects sure needs this. I know 5.4 isnt in LTS anymore, but it still is used a lot, IDK why would you want to limit your user base because of a version which changes nothing of significance AFAIK.

There are some differences in Symfony core packages, just check history for BinaryFileWrapper that this package adds. They are subtle, but I would like the behaviour to be as similar as possible.

Then there are differences for supported packages like Sentry.

Anyway, Symfony 6.4 will be supported in the next release (this weekend?)

Symfony 5.4 is almost out of question for me. It's EoL and receiving only security updates. If you want to use RoadRunner on old Symfony, you are not solving your problems, only prolonging the lifespan of that old system. Symfony 5.4 has minimum PHP version of 7.2. None of RoadRunner packages work with that as they use PHP 8.1 as minimum AFAIK. I am though aware that many Symfony 4.4 and 5.4 projects are still using/stuck with PHP 7.4, so that would be the target to go for, but as I mentioned, other packages simply don't support it.

TLDR Symfony 5.4: I could _technically_ support it on PHP 8.1. I will look into that, but no promises, as it has too many missing features.

**Comment by moshkov-konstantin** [comment]
Don't waste your time. Projects should be designed so that they don't depend on the framework, much less its version.

**Comment by Zylius** [comment]
Hello,

Thank you for response. I am aware that supporting Symfony 5 can be kinda awkward. From a large project perspective the situation is that upgrading PHP is much easier than Symfony, all our services run PHP 8.3. Symfony is getting updated much more slowly.

Using RoadRunner on old Symfony is fine.  TBQH framework doesn't matter for RoadRunner, neither does its version. I personally wrote an in-house bundle for handling Temporal with RoadRunner back in 2021. We didn't have new Symfony back then. It is still going strong with the newest RoadRunner packages, binary, temporal SDKs etcetra. There is a reason why [temporal](https://github.com/temporalio/sdk-php/blob/master/composer.json#L41) and [roadrunner](https://github.com/roadrunner-php/cli/blob/2.x/composer.json#L37) still supports Symfony/* ^5 packages.

In the mean time on worst case scenario I think I'll just fork it. I've already forked baldinof for this exact reason, everything seems to be working with 5.x.

In the end what I need is proper http worker handling, so I'll end up writing that myself cause upgrading projects will take longer time. Since we're moving our APIs from FPM to RoadRunner, the savings we could get is not worth losing over a version constraint :D

**Comment by FluffyDiscord** [comment]
> Don't waste your time. Projects should be designed so that they don't depend on the framework, much less its version.

I will entertain you only once. So read the following paragraphs carefully.

I am maintaining this package for myself, for my use cases.

Single person, in their spare time, without any funding.

It's open on Github only because I can't be bothered to manage it in my private Git.

You are welcome to stop bitching about package requiring specific framework versions and forking or writing your own stuff. There is also option to fund me. Do that so I can have more time for this package.

**Comment by Zylius** [comment]
Welp I'm not bitching, weird you would see it that way. I'm trying to, of course, use it for my own needs, but if you see every question like "Hello could you please add support for 5.4 and 6.4?" as bitching or condescending, it might hurt your own public project. 

Especially since you opened the issue about version support. Weird it's listed as an option for using with roadrunner in roadrunner docs if it's only for your needs, as you say. That might need to be fixed.

As for everything else it's your choice :D

**Comment by FluffyDiscord** [comment]
> Welp I'm not bitching, weird you would see it that way. I'm trying to, of course, use it for my own needs, but if you see every question like "Hello could you please add support for 5.4 and 6.4?" as bitching or condescending, it might hurt your own public project.
> 
> Especially since you opened the issue about version support. Weird it's listed as an option for using with roadrunner in roadrunner docs if it's only for your needs, as you say. That might need to be fixed.
> 
> As for everything else it's your choice :D

I wasn't replying to you ;) (check the quote) and I think I answered you positively - that support for symfony 6.4 will be done as early as possible, and thst I might also add support for 5.4

**Comment by Zylius** [comment]
Oh sorry,  I've read the quote
 
```
You are welcome to stop bitching about package requiring specific framework versions and forking or writing your own stuff. There is also option to fund me. Do that so I can have more time for this package.
```

And I thought it was referencing my comment :D

**Comment by FluffyDiscord** [comment]
@Zylius package should be now compatible with Symfony 5.4 and 6.4. I have tested it on empty/new Symfony web skeleton since I do not have any projects on these Symfony versions, so I would appreciate if you could report back or create PR for issues you find.

**Comment by Zylius** [comment]
Thank you very much :) I'll get back to you with an MR if I find any issues :)
